### PR TITLE
feat: implement homepage ui

### PR DESCRIPTION
**Pull request by PainterPuppets**
Implemented a homepage page based on the ui design.

image

### CODE REVIEW

File: tailwind.config.ts [hunk 1]
```diff
@@ -11,11 +11,13 @@ const config: Config = {
...
         'primary-color': 'var(--primary-color)',
         'secondary-color': 'var(--secondary-color)',
+        'lighter-color': 'var(--lighter-color)',
+        'highlight-color': 'var(--highlight-color)',
       },
       backgroundImage: {
         'gradient-radial': 'radial-gradient(var(--tw-gradient-stops))',
         'gradient-conic': 'conic-gradient(from 180deg at 50% 50%, var(--tw-gradient-stops))',
-      },
+      },
```
Comment (Keith-CY) [inline tailwind.config.ts]:
```suggestion
      },
```

File: app/utils/index.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    if (bigValue.isNegative()) {
+      return 0 - Math.floor(num * 10 ** decimal) / 10 ** decimal
+    }
+    return Math.floor(num * 10 ** decimal) / 10 ** decimal
+  }
+  if (bigValue.isNegative()) {
+    return 0 - Math.floor(num)
+  }
+  return Math.floor(num)
+}
```
Comment (Keith-CY) [inline app/utils/index.ts]:
```suggestion
}

```

File: app/utils/number.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import BigNumber from 'bignumber.js'                                                        
+
+export const localeNumberString = (value: BigNumber | string | number): string => {
+  if (!value) return '0'
+  const origin = typeof value === 'string' || typeof value === 'number' ? new BigNumber(value) : value
+  const bigValue = origin.abs()
+  if (bigValue.isNaN()) {
+    return '0'
+  }
+  if (bigValue.isLessThan(1) && bigValue.abs().isGreaterThan(0)) {
```
Comment (Keith-CY) [inline app/utils/number.ts]: `bigValue` is non-negative

Comment (Keith-CY) [reply app/utils/number.ts]: Why returned without `+/-` sign while the sign is returned along with the value in the following logic

File: app/utils/number.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    return `${value}`
+  }
+  let text = bigValue.toString(10)
+  const pointIndex = text.indexOf('.')
+  let offset = pointIndex === -1 ? text.length : pointIndex
+  while (offset > 3) {
```
Comment (Keith-CY) [inline app/utils/number.ts]: `bignumber` has its own formatter

Ref:
- toFormat: https://mikemcl.github.io/bignumber.js/#toFor

File: app/utils/index.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import BigNumber from 'bignumber.js'
+
+export const shannonToCkb = (value: BigNumber | string | number): string => {
```
Comment (Keith-CY) [inline app/utils/index.ts]: It can be done by `parseUDTAmount`, and the name `parseUDTAmount` limits its usage

File: app/utils/index.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  if (num.abs().isLessThan(new BigNumber('1e-6'))) {
+    if (bigValue.mod(10).isEqualTo(0)) {
+      return num.toFixed(7)
+    }
+    return num.toFixed(8)
+  }
+  return `${num}`
+}
+
+export const shannonToCkbDecimal = (value: BigNumber | string | number, decimal?: number) => {
```
Comment (Keith-CY) [inline app/utils/index.ts]: What's the difference between `shannonToCkb` and `shannonToCkbDecimal`

**Comment by Keith-CY** [comment]
Any update on this PR

**Comment by PainterPuppets** [comment]
> Any update on this PR

I am optimizing the `tabs` and complete the `history` and `form` part. 
And `utils` function is from the `explorer-ui` project. I'm wondering if it would be better to have a separate repo for these utils functions.

**Comment by Keith-CY** [comment]
> > Any update on this PR
> 
> I am optimizing the `tabs` and complete the `history` and `form` part. And `utils` function is from the `explorer-ui` project. I'm wondering if it would be better to have a separate repo for these utils functions.

It would be a good habit to keep PR small, but you may decide it by yourself in this project because now only we two are maintaining it.

**Comment by Keith-CY** [comment]
Any update on this @PainterPuppets

**Comment by PainterPuppets** [comment]
Fixed it according to commit

**Review by Keith-CY — APPROVED** [review]

**Pull request merged**